hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieMergeOnReadRDD.scala

```diff
@@ -116,7 +125,33 @@ class HoodieMergeOnReadRDD(@transient sc: SparkContext,
       Option(TaskContext.get()).foreach(_.addTaskCompletionListener[Unit](_ => iter.asInstanceOf[Closeable].close()))
     }
 
-    iter
+    val commitTimeMetadataFieldIdx = requiredSchema.structTypeSchema.fieldNames.indexOf(HoodieRecord.COMMIT_TIME_METADATA_FIELD)
+    val needsFiltering = commitTimeMetadataFieldIdx >= 0 && !StringUtils.isNullOrEmpty(startTimestamp) && !StringUtils.isNullOrEmpty(endTimestamp)
+    if (needsFiltering) {
```

Comment:
Don't thinking adding `filter` function to `HoodieMergeOnReadRDD` is a appropriate way, it's a lower level to solely read splits. what abt we filter `RDD` in `buildScan`(`HoodieBaseRelation`), this way both `COW` and `MOR` can benefit from it.

Reply:
cow table has different filtering strategy, if possible, can you write a test case for it and we can merge it fast.

Reply:
I also tried to add a filter to the `buildScan` method of `HoodieBaseRelation` at the beginning, but there was a problem that the class was `not serializable`, so I moved to the `HoodieMergeOnReadRDD` class.